src/util/SqlTagUtils.ts

```diff
@@ -0,0 +1,61 @@
+import { Driver } from "../driver/Driver"
+import dedent from "dedent"
+
+interface BuildSqlTagParams {
+    driver: Driver
+    strings: TemplateStringsArray
+    expressions: unknown[]
+}
+
+export function buildSqlTag({
+    driver,
+    strings,
+    expressions,
+}: BuildSqlTagParams): { query: string; parameters: unknown[] } {
+    let query = ""
+    const parameters: unknown[] = []
+    let idx = 0
+
+    for (const [expressionIdx, expression] of expressions.entries()) {
+        query += strings[expressionIdx]
+
+        if (expression === null) {
+            query += "NULL"
+            continue
+        }
+
+        if (Array.isArray(expression)) {
+            if (expression.length === 0) {
+                query += "NULL"
+                continue
+            }
+
+            const arrayParams = expression.map(() => {
+                return driver.createParameter(`param_${idx + 1}`, idx++)
+            })
+
+            query += arrayParams.join(", ")
+            parameters.push(...expression.map((e) => toParameter(e)))
+
+            continue
+        }
+
+        query += driver.createParameter(`param_${idx + 1}`, idx++)
+
+        parameters.push(toParameter(expression))
+    }
+
+    query += strings[strings.length - 1]
+
+    query = dedent(query)
+
+    return { query, parameters }
+}
+
+function toParameter(expression: unknown) {
```

Comment:
probably we don't need to bother with this, the user can call the function manually 🤷🏻

Reply:
It matters because of https://github.com/typeorm/typeorm/pull/11432#discussion_r2069537794

See the [following test](https://github.com/typeorm/typeorm/pull/11432/files#diff-1f7a118eed75b848ed6280db578e0af5e685221db97ce2cbe201a300bd83d689R208)